Supervisor feedback (#81)
chhoumann/p7: report/sections/Chapter 03 - Speecification/Specification.tex

\subsection*{Use case: Show statistics over exercises}
A Lecturer should be able to view statistics about exercises, for example how many attempts were needed by the Students to complete it or the time taken, in order to determine if the difficulty is sufficient. To do this the system needs to log the information as the users attempt to complete the exercises.

\subsection*{Use case: View solutions}
The Lecturer can view all submitted solutions, to see if the students use the correct approach.
The Students should be able to view their own solutions and attempts for the exercises.
To do this a database keeping a track of the code used for each attempt submitted is needed. 

\subsection*{Use case: Best solutions} 
The Lecturer should be able to select code solutions submitted by the Students to publish to all other Students allowing them to view exercises discussed during class.

With the use cases that are contained in our scope described, we need a way to prioritize them. This then allows us to make a prioritized list of the features that are needed in-order to fulfill our use cases. 

\section{MoSCoW}

> To gain an overview of which features the solution must, should, could and wont contain we perform a MoSCoW analysis.

Reviewer: ```suggestion
To gain an overview of which features the solution must, should, could and won't contain we perform a MoSCoW analysis.
```

A MoSCoW analysis is split into four categories, Must have, Should have, Could have and Would have. Each feature will then be put into a category based on how important it is for the system. 

> Here table \ref{tab:MOSCOW} provides an overview of where the different needed features were placed in order to gain a working solution. 

Reviewer: ```suggestion
Table \ref{tab:MOSCOW} shows the feature prioritizations for a working solution. 
```
